### PR TITLE
Clear the current task panel

### PR DESCRIPTION
**Pull request by wxy1203**
### Changes ###
- task command: task clear
- clear the current task panel and create another new empty task panel

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package seedu.address.logic.parser.task;
+
+import seedu.address.logic.commands.task.ClearTaskCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new MarkTaskCommand object
+ */
+public class ClearTaskCommandParser implements Parser<ClearTaskCommand> {
```
Comment (yilinzyl) [inline src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java]: This class seems pretty redundant! Could consider just returning new ClearTaskCommand() in TaskPanelParser.java. (Can refer to AddressBookParser.java for reference)

File: src/test/java/seedu/address/logic/parser/task/ClearTaskCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * same path through the ClearTaskCommand, and therefore we test only one of them.
+ * The path variation for those two cases occur inside the ParserUtil, and
+ * therefore should be covered by the ParserUtilTest.
+ */
+public class ClearTaskCommandParserTest {
+
+    private ClearTaskCommandParser parser = new ClearTaskCommandParser();
+
+    @Test
+    public void parse_validArgs_returnsDeleteCommand() {
```
Comment (yilinzyl) [inline src/test/java/seedu/address/logic/parser/task/ClearTaskCommandParserTest.java]:
```suggestion
    public void parse_validArgs_returnsClearTaskCommand() {
```

**Review by yilinzyl — APPROVED** [review]
LGTM!

File: src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package seedu.address.logic.parser.task;
+
+import seedu.address.logic.commands.task.ClearTaskCommand;
+import seedu.address.logic.parser.Parser;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new MarkTaskCommand object
```
Comment (kianseong) [inline src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java]: Could we change this to ClearTaskCommand?

File: src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ */
+public class ClearTaskCommandParser implements Parser<ClearTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the MarkTaskCommand
+     * and returns an MarkTaskCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
```
Comment (kianseong) [inline src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java]: JavaDocs here as well.

**Review by kianseong — APPROVED** [review]
LGTM other than some minor JavaDocs issues

File: src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java (hunk 1 above)
Comment (wxy1203) [reply src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java]: Finished

File: src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java (hunk 4 above)
Comment (wxy1203) [reply src/main/java/seedu/address/logic/parser/task/ClearTaskCommandParser.java]: No parser for clear task command any more.

**Pull request merged**